CountToMillion/SharedObject.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <string>
+#include <Windows.h>
+#include "WinAPIException.h"
+
+
+/*
+ * This class represents a shared object.
+ * It uses a mutex to make sure only on thread can access it.
+ *
+ */
+class SharedObject
+{
+public:
+	/*
+	 * The constructor - initiazes obj, and creates the mutex.
+	 * 
+	 * @param obj - the object to copy
+	 * @param name - the mutex's name
+	 */
+	SharedObject(DWORD obj, const std::wstring& name);
+
+	/*
+	 * The shared value getter. This fucntion waits for the mutex to be signaled.
+	 *
+	 * @return - The shared value
+	 */
+	DWORD getValue() const;
+
+	/*
+	 * The shared value setter. This fucntion waits for the mutex to be signaled.
+	 *
+	 * @param newVal - The new shared value
+	 */
+	void setValue(DWORD newVal);
+
+	// Rule of 3 - delete copy assignment and copy ctor.
```

Comment:
Rule of 5 - delete the move ctor & assignment operator if they're not used

- [ ] Rule of 5